ui/src/app/zones/views/ZoneDetails/ZoneDetailsForm/ZoneDetailsForm.tsx

```diff
@@ -21,13 +22,29 @@ export type CreateZoneValues = {
 
 const ZoneForm = ({ id, closeForm }: Props): JSX.Element | null => {
   const dispatch = useDispatch();
-  const errors = useSelector(zoneSelectors.errors);
-  const saved = useSelector(zoneSelectors.saved);
-  const saving = useSelector(zoneSelectors.saving);
   const cleanup = useCallback(() => zoneActions.cleanup(), []);
   const zone = useSelector((state: RootState) =>
     zoneSelectors.getById(state, Number(id))
   );
+  const errors = useSelector((state: RootState) =>
+    zoneSelectors.getLatestActionError(state, ZONE_ACTIONS.create)
+  );
+  const updated = useSelector((state: RootState) =>
+    zoneSelectors.getModelActionState(
+      state,
+      id,
+      ZONE_ACTIONS.update,
+      ACTION_STATUS.successful
+    )
+  );
+  const updating = useSelector((state: RootState) =>
+    zoneSelectors.getModelActionState(
+      state,
+      id,
+      ZONE_ACTIONS.update,
+      ACTION_STATUS.processing
+    )
+  );
```

Comment:
When using a single status string this could be simplified like below:
```jsx
const status: 'idle' | 'processing' | 'success' | 'error' =
 useSelector(zoneSelectors.getModelActionstate(state, id, ZONE_ACTIONS.update)
```

Reply:
That's a great point! I much prefer this alternative, thanks